framer/Context.coffee

```diff
@@ -1,122 +1,241 @@
+		@_animations.map (animation) -> animation.stop(true)
 
-		@_rootElement = document.createElement("div")
-		@_rootElement.id = "FramerContextRoot-#{@_name}"
-		@_rootElement.classList.add("framerContext")
 
-		if @_parentLayer
-			@_appendRootElement()
-		else
-			Utils.domComplete(@_appendRootElement)
+	# Timers
+	@define "timers", get: -> _.clone(@_timers)
+	
+	addTimer: (timer) ->
+		return if timer in @_timers
+		@_timers.push(timer)
+		
+	removeTimer: (timer) ->
+		@_timers = _.without(@_timers, timer)
+	
+	resetTimers: ->
+		@_timers.map(window.clearTimeout) if @_timers
+		@_timers = []
 
-	_appendRootElement: =>
-		parentElement = @_parentLayer?._element
-		parentElement ?= document.body
-		parentElement.appendChild(@_rootElement)
 
-	run: (f) ->
+	# Intervals
+	@define "intervals", get: -> _.clone(@_intervals)
+	
+	addInterval: (interval) ->
+		return if interval in @_intervals
+		@_intervals.push(interval)
+		
+	removeInterval: (interval) ->
+		@_intervals = _.without(@_intervals, interval)
+	
+	resetIntervals: ->
+		@_intervals.map(window.clearInterval) if @_intervals
+		@_intervals = []
+
+
+	##############################################################
+	# Run
+
+	run: (fn) ->
 		previousContext = Framer.CurrentContext
 		Framer.CurrentContext = @
-		f()
+		fn()
 		Framer.CurrentContext = previousContext
 
+
+	##############################################################
+	# Freezing
+
+	freeze: ->
+
+		if @_frozenEvents
+			throw new Error "Context is already frozen"
+
+		@_frozenEvents = {}
+
+		for layer in @_layers
+
+			layerListeners = layer.listeners()
+			layerId = @_layers.indexOf(layer)
+			layer.removeAllListeners()
+
+			@_frozenEvents[layerId] = layerListeners
+			
+		@stopAnimations()
+
+		# TODO: It would be nice to continue at least intervals after a resume
+		@resetTimers()
+		@resetIntervals()
+
+	resume: ->
+
+		if not @_frozenEvents
+			throw new Error "Context is not frozen, cannot resume"
+
+		for layerId, events of @_frozenEvents
+			layer = @_layers[layerId]
+			for eventName, listeners of events
+				for listener in listeners
+					layer.on(eventName, listener)
+
+		delete @_frozenEvents
+
+
+	##############################################################
+	# DOM
+
+	_createDOMEventManager: ->
+		@domEventManager?.reset()
+		@domEventManager = new DOMEventManager
+
+	_destroyRootElement: ->
+
+		if @_element?.parentNode
+			@_element.parentNode.removeChild(@_element)
+
+		if @__pendingElementAppend
+			Utils.domCompleteCancel(@__pendingElementAppend)
+			@__pendingElementAppend = null
+
+		@_element = null
+
+	_createRootElement: ->
+
+		@_destroyRootElement()
+
+		@_element = document.createElement("div")
+		@_element.id = "FramerContextRoot-#{@_name}"
+		@_element.classList.add("framerContext")
+
+		@__pendingElementAppend = =>
+			parentElement = @_parent?._element
+			parentElement ?= document.body
+			parentElement.appendChild(@_element)
+
+		Utils.domComplete(@__pendingElementAppend)
+
+
```

Comment:
@koenbok Could this get some comments explaining what's going on? A context has its own `div`? Why does it need that when it's also attached to a Layer?